src/vmm/src/cpu_config/aarch64/mod.rs

```diff
@@ -20,14 +22,27 @@ pub struct Error(#[from] pub ArchError);
 #[derive(Debug, Default, Clone, PartialEq, Eq)]
 pub struct CpuConfiguration {
     /// Vector of CPU registers
-    pub regs: Vec<Aarch64Register>,
+    pub regs: Aarch64RegisterVec,
 }
 
 impl CpuConfiguration {
     /// Creates new guest CPU config based on the provided template
     pub fn apply_template(mut self, template: &CustomCpuTemplate) -> Result<Self, Error> {
-        for (modifier, reg) in template.reg_modifiers.iter().zip(self.regs.iter_mut()) {
-            reg.value = modifier.bitmap.apply(reg.value);
+        for (modifier, mut reg) in template.reg_modifiers.iter().zip(self.regs.iter_mut()) {
+            match reg.size() {
+                RegSize::U32 => {
+                    *reg.value_mut::<u32>() =
+                        modifier.bitmap.apply(u128::from(*reg.value::<u32>())) as u32;
+                }
+                RegSize::U64 => {
+                    *reg.value_mut::<u64>() =
+                        modifier.bitmap.apply(u128::from(*reg.value::<u64>())) as u64;
+                }
+                RegSize::U128 => {
+                    *reg.value_mut::<u128>() = modifier.bitmap.apply(*reg.value::<u128>());
+                }
+                _ => unreachable!("Only 32, 64 and 128 bit wide registers are supported"),
```

Comment:
are we going to support 256 when/if we add support for SVE?

Reply:
Well... we definitely can support it, but does it really make sense to change SVE registers? Anyway, currently we are limited by `u128` in cpu templates.

Reply:
fair enough